File: planner/app/src/main/java/com/sunilson/firenote/dialogs/SuperDialog.java
package com.sunilson.firenote.dialogs;

import android.app.Activity;
import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.sunilson.firenote.R;
import com.sunilson.firenote.activities.BaseElementActivity;

/**
 * @author Linus Weiss
 *
 */

/**
 * Base class of all dialogs
 */
public class SuperDialog extends DialogFragment {

    protected TextView titleText;
    protected LayoutInflater inflater;
    protected AlertDialog.Builder builder;
    protected Activity activity;
    protected View title;

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {

        //Basic functionality of AlertDialog. Same for all dialogs
        builder = new AlertDialog.Builder(getActivity());
        activity = getActivity();
        inflater = getActivity().getLayoutInflater();
        title = inflater.inflate(R.layout.alertdialog_custom_title, null);
        titleText = (TextView) title.findViewById(R.id.dialog_title);

        //Check if we are in an Element. If yes, set color of dialog to color of Element
        if (activity instanceof BaseElementActivity) {
            (title.findViewById(R.id.dialog_title_container)).setBackgroundColor(((BaseElementActivity)activity).getElementColor());
        }

        return super.onCreateDialog(savedInstanceState);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        //Apply open and close animation to Dialog
        if (getDialog().getWindow() != null) {
            getDialog().getWindow().getAttributes().windowAnimations = R.style.dialogAnimation;
        }
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/dialogs/SuperDialog.kt
package com.sunilson.firenote.presentation.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.app.DialogFragment
import android.os.Bundle
import android.view.View
import com.sunilson.firenote.R
import com.sunilson.firenote.presentation.shared.activities.BaseElementActivity
import kotlinx.android.synthetic.main.alertdialog_custom_title.view.*

abstract class SuperDialog : DialogFragment() {

    protected lateinit var titleView: View
    protected lateinit var builder: AlertDialog.Builder

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        builder = AlertDialog.Builder(activity)
        titleView = activity.layoutInflater.inflate(R.layout.alertdialog_custom_title, null)
        if (activity is BaseElementActivity) titleView.dialog_title_container.setBackgroundColor((activity as BaseElementActivity).elementColor)
        return super.onCreateDialog(savedInstanceState)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if(dialog.window != null) dialog.window.attributes.windowAnimations = R.style.dialogAnimation
    }
}